javascript/house-style.md

```diff
@@ -195,6 +195,46 @@ We don't do this:
 const uniqueIDs = new Set([1, 2, 1, 2]);
 ```
 
+#### JS Doc
+
+Although your JavaScript code should be self-documenting, we highly recommend
+that you to document your JavasScript application or library using the
+Standardised [JS Doc](https://jsdoc.app/) format.
+
+This format is leveraged by the Language Server Protocol (LSP), which is
+supported in numerous editors and Integrated Development Environments (IDE)
+nowadays.  
+Thanks to LSP you get short lines of documentation right from the
+invocation context. This helps you code quickly, while avoiding obvious errors.
+
+For functions for instance, you get:
+- What it is intended to do
+- The type, default value, and description of its parameters
+- Which parameters are optional
+- The function returned value type and description
+
+Bonus: There are tools to generate a documentation website for your application
+or library if it uses JS Doc.
+
+A concrete and furnished example:
+
+```js
+/**
+ * Get a list of books from given a library.
+ *
+ * @param {number} library - Id of the library.
+ * @param {number} [limit=10] - Optional maximum of books expected to be returned, defaults to 10.
+ * @param {object} [filters={}] - Optional filters, defaults to no filters.
+ * @param {string} [filters.subject] - A string to match up the subject of the book.
+ * @param {number} [filters.publicationyear] - Year the book was published.
+ *
+ * @returns {object[]} list of books from that library matching the optionally given filters.
```

Comment:
For consistency with the param lines:

```suggestion
 * @returns {object[]} List of books from that library matching the optionally given filters.
```

Or perhaps return a list of ids here, or the reader might wonder why favouritism has been shown to `filters` or assume that describing an array of objects isn't possible:

```suggestion
 * @returns {string[]} List of book ids matching the optionally given filters.
```

Then again, that might make the name of the function imprecise.  A dilemma.